src/ts/settlement.ts

```diff
@@ -260,25 +258,57 @@ export class SettlementEncoder {
   }
 
   /**
-   * Gets the encoded interactions for the specified stage as a hex-encoded
-   * string.
+   * Gets all encoded interactions for all stages.
+   *
+   * Note that order refund interactions are included as post-interactions.
    */
   public get interactions(): [Interaction[], Interaction[], Interaction[]] {
     return [
       this._interactions[InteractionStage.PRE].slice(),
       this._interactions[InteractionStage.INTRA].slice(),
-      this._interactions[InteractionStage.POST].slice(),
+      [
+        ...this._interactions[InteractionStage.POST],
+        ...this.encodedOrderRefunds,
+      ],
     ];
   }
 
   /**
-   * Gets the currently encoded order UIDs for gas refunds.
+   * Gets the order refunds encoded as interactions.
    */
-  public get orderRefunds(): OrderRefunds {
-    return {
-      filledAmounts: this._orderRefunds.filledAmounts.slice(),
-      preSignatures: this._orderRefunds.preSignatures.slice(),
-    };
+  public get encodedOrderRefunds(): Interaction[] {
+    const { filledAmounts, preSignatures } = this._orderRefunds;
+    if (filledAmounts.length + preSignatures.length === 0) {
+      return [];
+    }
+
+    const settlement = this.domain.verifyingContract;
+    if (settlement === undefined) {
+      throw new Error("domain missing settlement contract address");
+    }
+
+    // NOTE: Avoid importing the full GPv2Settlement contract artifact just for
+    // a tiny snippet of the ABI. Unit and integration tests will catch any
+    // issues that may arise from this definition becoming out of date.
+    const iface = new ethers.utils.Interface([
+      "function freeFilledAmountStorage(bytes[] orderUids)",
+      "function freePreSignatureStorage(bytes[] orderUids)",
+    ]);
+
+    const interactions = [];
+    for (const [functionName, orderUids] of [
+      ["freeFilledAmountStorage", filledAmounts] as const,
+      ["freePreSignatureStorage", preSignatures] as const,
```

Comment:
`as const` makes it so the type is `[string, BytesLike[]]` instead of `(string | BytesLike[])[]`. The `as const` is necessary for tuple type inference.